pkg/scheduler/plugins/capacity/capacity.go

```diff
@@ -165,25 +165,36 @@ func (cp *capacityPlugin) OnSessionOpen(ssn *framework.Session) {
 		return !overused
 	})
 
-	ssn.AddAllocatableFn(cp.Name(), func(queue *api.QueueInfo, candidate *api.TaskInfo) bool {
-		if !readyToSchedule {
-			klog.V(3).Infof("Capacity plugin failed to check queue's hierarchical structure!")
-			return false
-		}
-
-		if hierarchyEnabled && !cp.isLeafQueue(queue.UID) {
-			return false
-		}
+	queueAllocatable := func(queue *api.QueueInfo, candidate *api.TaskInfo) bool {
 		attr := cp.queueOpts[queue.UID]
-
 		futureUsed := attr.allocated.Clone().Add(candidate.Resreq)
 		allocatable := futureUsed.LessEqualWithDimension(attr.realCapability, candidate.Resreq)
 		if !allocatable {
 			klog.V(3).Infof("Queue <%v>: realCapability <%v>, allocated <%v>; Candidate <%v>: resource request <%v>",
 				queue.Name, attr.realCapability, attr.allocated, candidate.Name, candidate.Resreq)
 		}
-
 		return allocatable
+	}
+
+	ssn.AddAllocatableFn(cp.Name(), func(queue *api.QueueInfo, candidate *api.TaskInfo) bool {
+		if !readyToSchedule {
+			klog.V(3).Infof("Capacity plugin failed to check queue's hierarchical structure!")
+			return false
+		}
+		if hierarchyEnabled {
+			if !cp.isLeafQueue(queue.UID) {
+				return false
+			}
+			allocatable := queueAllocatable(queue, candidate)
+			for _, parentID := range cp.queueOpts[queue.UID].parents {
+				if !queueAllocatable(ssn.Queues[parentID], candidate) && allocatable {
```

Comment:
> Why don't return false directly if allocatable is false?

Because I think we prefer to print all the logs that come from queues whose resources are insufficient.
Returning earlier can yield better performance if the leaf queue has too many parents, but our tree is shallow in most cases.

Reply:
logs and performance are both improtant... 
And I am not sure which aspect we should value more between these two...